### PR TITLE
feat: Upgrade Node Package Manager lock file `package-lock.json` to version 2

### LINKED ISSUE #8274: Remove CI check "NPM lock file version" and regenerate `package-lock.json` as v2
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

Removal of Node 12 support allows to safely move to package-lock.json v2.

### Feature / Enhancement Description

Remove CI check "NPM lock file version" and regenerate package-lock.json as v2

### Example Use Case
n/a

### Alternatives / Workarounds
n/a

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Lockfile should be v2 with Node 16

Related issue: #8274
Closes: #8274

### Approach

Generates lockfile

### TODOs before merging

- [x] Add tests
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -116,7 +116,7 @@ jobs:
       - name: Check NPM lock file version
         uses: mansona/npm-lockfile-version@v1
         with:
-          version: 1
+          version: 2
```
Comment (mtrezza) [inline .github/workflows/ci.yml:140]: Good idea to actually keep the version check

**Comment by mtrezza** [comment]
How is this getting along? Seems to be quite an effort looking at the number of commits

**Comment by mtrezza** [comment]
@dblythy Did this get stuck for some reason?

**Comment by dblythy** [comment]
The main problem is that lockfile v2 is much more strict with dependency mismatches, so there is some refactoring and updates that are required to ensure that `npm install` works. Are we still approaching this update even though we still support Node 14?

**Comment by mtrezza** [comment]
We could stay on v1 and exclude this from the initial Parse Server 6 release.

The only argument I see is that the majority of deployments are likely running or will soon run on Node >=16, since Node 14 LTS ends in April. The deployment process will be slower because of the one-time conversion of the lock file. It's also a resource waste. If we take 5.000 Parse Server installations per day (npm stats) and the one-time lock file conversion taking 30 seconds on avg., that would be 42 hours per day or 15,200 hours until Parse Server 7 release in 2024 of wasted compute time.

Lockfile v2 is backwards compatible with v1, but for anyone running custom scripts on the lockfile (like our CI does), this could be a breaking change. But it's a gray zone, we could also merge this anytime throughout the year.

I'll move this from "required" to "optional" in the [tracking list](https://github.com/parse-community/parse-server/issues/8225).

**Comment by dblythy** [comment]
I think it's worth migrating to V2 as it also is more restrictive of version conflicts, which can affect Node 14 as well.

**Review by mtrezza — APPROVED** [review]

**Comment by mtrezza** [comment]
Nice! It seems only deprecation warnings are left now, no more peer dependency warnings.

Rerunning CI

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-alpha.17](https://github.com/parse-community/parse-server/releases/tag/6.0.0-alpha.17)

**Comment by Moumouls** [comment]
So nice to be on V2, thanks @dblythy, @mtrezza !

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/6.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0](https://github.com/parse-community/parse-server/releases/tag/6.0.0)

**Pull request merged**